src/bower_components/apiclient/connectionmanager.js

```diff
@@ -1,236 +1,393 @@
+
+    var defaultTimeout = 2e4;
+    var ConnectionMode = {
+        Local: 0,
+        Remote: 1,
+        Manual: 2
+    };
+
+    var ConnectionManager = function (credentialProvider, appName, appVersion, deviceName, deviceId, capabilities, devicePixelRatio) {
+
+        //ToDo: Remove
+        /*function onConnectUserSignIn(user) {
+            connectUser = user;
+            events.trigger(self, "connectusersignedin", [user]);
+        }*/
+
+        function onAuthenticated(apiClient, result, options, saveCredentials) {
+            var credentials = credentialProvider.credentials();
+            var servers = credentials.Servers.filter(function (s) {
+                return s.Id === result.ServerId;
+            });
+            var server = servers.length ? servers[0] : apiClient.serverInfo();
+
+            if (false !== options.updateDateLastAccessed) {
+                server.DateLastAccessed = new Date().getTime();
+            }
+
+            server.Id = result.ServerId;
+
+            if (saveCredentials) {
+                server.UserId = result.User.Id;
+                server.AccessToken = result.AccessToken;
+            } else {
+                server.UserId = null;
+                server.AccessToken = null;
+            }
+
+            credentialProvider.addOrUpdateServer(credentials.Servers, server);
+            credentialProvider.credentials(credentials);
+            apiClient.enableAutomaticBitrateDetection = options.enableAutomaticBitrateDetection;
+            apiClient.serverInfo(server);
+            afterConnected(apiClient, options);
+            return onLocalUserSignIn(server, apiClient.serverAddress(), result.User);
+        }
+
+        function afterConnected(apiClient, options) {
+            options = options || {};
+
+            if (false !== options.reportCapabilities) {
+                apiClient.reportCapabilities(capabilities);
             }
 
-            function afterConnected(apiClient, options) {
-                options = options || {}, !1 !== options.reportCapabilities && apiClient.reportCapabilities(capabilities), apiClient.enableAutomaticBitrateDetection = options.enableAutomaticBitrateDetection, !1 !== options.enableWebSocket && (console.log("calling apiClient.ensureWebSocket"), apiClient.ensureWebSocket())
+            apiClient.enableAutomaticBitrateDetection = options.enableAutomaticBitrateDetection;
+
+            if (false !== options.enableWebSocket) {
+                console.log("calling apiClient.ensureWebSocket");
+                apiClient.ensureWebSocket();
             }
+        }
+
+        function onLocalUserSignIn(server, serverUrl, user) {
+            self._getOrAddApiClient(server, serverUrl);
+
+            return (self.onLocalUserSignedIn ? self.onLocalUserSignedIn.call(self, user) : Promise.resolve()).then(function () {
```

Comment:
looks ugly, can this be improved somehow?

Reply:
https://github.com/jellyfin/jellyfin-web/pull/520/commits/1e3b03dde10c1cc27a40cdc745b63a1c9a1ca380